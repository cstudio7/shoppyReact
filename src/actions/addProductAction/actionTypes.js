const actionTypes = {
  ADD_PRODUCT_LOADING: 'ADD_PRODUCT_LOADING',
  ADD_PRODUCT_SUCCESS: 'ADD_PRODUCT_SUCCESS',
  ADD_PRODUCT_FAILURE: 'ADD_PRODUCT_FAILURE'
};

export default actionTypes;
