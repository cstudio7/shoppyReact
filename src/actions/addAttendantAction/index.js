import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/addAttendantRequest';

const { ADD_ATTENDANT_FAILURE, ADD_ATTENDANT_LOADING, ADD_ATTENDANT_SUCCESS } = actionTypes;

const addAttendantLoading = type => ({
  type
});

const addAttendantSuccess = payload => ({
  type: ADD_ATTENDANT_SUCCESS,
  payload
});

const addAttendantFailure = payload => ({
  type: ADD_ATTENDANT_FAILURE,
  payload
});

const addStoreAttendant = attendantDetails => async (dispatch) => {
  try {
    dispatch(addAttendantLoading(ADD_ATTENDANT_LOADING));
    const response = await request.addAttendantRequest(attendantDetails);
    dispatch(addAttendantSuccess(response.data.rows));
    toast.success('Attendant created successfully');
  } catch (error) {
    dispatch(addAttendantFailure(error.response.data.message));
  }
};

export { addStoreAttendant, addAttendantLoading, addAttendantSuccess, addAttendantFailure };
