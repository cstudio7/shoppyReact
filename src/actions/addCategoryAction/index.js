import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/addCategoryRequest';

const { ADD_CATEGORY_FAILURE, ADD_CATEGORY_LOADING, ADD_CATEGORY_SUCCESS } = actionTypes;

const addCategoryLoading = type => ({
  type
});

const addCategorySuccess = payload => ({
  type: ADD_CATEGORY_SUCCESS,
  payload
});

const addCategoryFailure = payload => ({
  type: ADD_CATEGORY_FAILURE,
  payload
});

const addCategory = categoryDetails => async (dispatch) => {
  try {
    dispatch(addCategoryLoading(ADD_CATEGORY_LOADING));
    const response = await request.addCategoryRequest(categoryDetails);
    dispatch(addCategorySuccess(response.data.rows));
    toast.success('Category created successfully');
  } catch (error) {
    dispatch(addCategoryFailure(error.response.data));
  }
};

export { addCategory, addCategoryLoading, addCategorySuccess, addCategoryFailure };
