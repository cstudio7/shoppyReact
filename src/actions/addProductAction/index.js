import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/addProductRequest';

const { ADD_PRODUCT_FAILURE, ADD_PRODUCT_LOADING, ADD_PRODUCT_SUCCESS } = actionTypes;

const addProductLoading = type => ({
  type
});

const addProductSuccess = payload => ({
  type: ADD_PRODUCT_SUCCESS,
  payload
});

const addProductFailure = payload => ({
  type: ADD_PRODUCT_FAILURE,
  payload
});

const addProduct = (productDetails, categoryId) => async (dispatch) => {
  try {
    dispatch(addProductLoading(ADD_PRODUCT_LOADING));
    const response = await request.addProductRequest(productDetails, categoryId);
    dispatch(addProductSuccess(response.data.rows));
    toast.success('Product created successfully');
  } catch (error) {
    dispatch(addProductFailure(error.response.data));
  }
};

export { addProduct, addProductLoading, addProductSuccess, addProductFailure };
