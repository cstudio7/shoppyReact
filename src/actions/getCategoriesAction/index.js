import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/getCategoriesRequest';

const { GET_CATEGORIES_FAILURE, GET_CATEGORIES_LOADING, GET_CATEGORIES_SUCCESS } = actionTypes;

const categoriesLoading = type => ({
  type
});

const categoriesSuccess = payload => ({
  type: GET_CATEGORIES_SUCCESS,
  payload
});

const categoriesFailure = payload => ({
  type: GET_CATEGORIES_FAILURE,
  payload
});

const productCategories = () => async (dispatch) => {
  try {
    dispatch(categoriesLoading(GET_CATEGORIES_LOADING));
    const response = await request.getCategoriesRequest();
    dispatch(categoriesSuccess(response.data.rows));
  } catch (error) {
    dispatch(categoriesFailure(error.response));
  }
};

export { productCategories, categoriesFailure, categoriesSuccess, categoriesLoading };
