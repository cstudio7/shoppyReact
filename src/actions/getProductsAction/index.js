import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/getProductsRequest';

const { GET_PRODUCTS_FAILURE, GET_PRODUCTS_LOADING, GET_PRODUCTS_SUCCESS } = actionTypes;

const getProductsLoading = type => ({
  type
});

const getProductsSuccess = payload => ({
  type: GET_PRODUCTS_SUCCESS,
  payload
});

const getProductsFailure = payload => ({
  type: GET_PRODUCTS_FAILURE,
  payload
});

const getAllProducts = () => async (dispatch) => {
  try {
    dispatch(getProductsLoading(GET_PRODUCTS_LOADING));
    const response = await request.getProductsRequest();

    dispatch(getProductsSuccess(response.data.rows));
  } catch (error) {
    dispatch(getProductsFailure(error.response));
  }
};

export { getProductsLoading, getProductsSuccess, getProductsFailure, getAllProducts };
