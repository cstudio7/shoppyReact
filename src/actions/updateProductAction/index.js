import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import request from '../../helpers/axiosHelper/updateProductRequest';

const { UPDATE_PRODUCT_FAILURE, UPDATE_PRODUCT_LOADING, UPDATE_PRODUCT_SUCCESS } = actionTypes;

const updateProductLoading = type => ({
  type
});

const updateProductSuccess = payload => ({
  type: UPDATE_PRODUCT_SUCCESS,
  payload
});

const updateProductFailure = payload => ({
  type: UPDATE_PRODUCT_FAILURE,
  payload
});

const updateProduct = (productDetails, categoryId, productId) => async (dispatch) => {
  try {
    dispatch(updateProductLoading(UPDATE_PRODUCT_LOADING));
    const response = await request.updateProductRequest(productDetails, categoryId, productId);
    dispatch(updateProductSuccess(response.data.Message));
    toast.success('Product updated successfully');
  } catch (error) {
    dispatch(updateProductFailure(error.response));
  }
};

export { updateProduct, updateProductFailure, updateProductLoading, updateProductSuccess };
