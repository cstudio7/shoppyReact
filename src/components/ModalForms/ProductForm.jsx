import React, { Component, Fragment } from 'react';
import { Button, Form, Select } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { productCategories } from '../../actions/getCategoriesAction';
import { updateProduct } from '../../actions/updateProductAction';
import { getAllProducts } from '../../actions/getProductsAction';
import { addProduct } from '../../actions/addProductAction';
import './ModalForms.scss';

class ProductForm extends Component {
  state = {
    category_id: this.props.details.categoryId || '',
    category_name: this.props.details.categoryName || '',
    productDetails: {
      name: this.props.details.nameToDisplay || '',
      image_url: this.props.details.imageUrl || '',
      description: this.props.details.description || '',
      price: this.props.details.productPrice || '',
      quantity: this.props.details.productQuantity || ''
    }
  };

  componentWillUnmount = () => {
    clearTimeout(this.state.timeoutId);
  };

  getAllCategories = () => {
    this.props.productCategories();
  };

  handleFormSubmit = (event) => {
    event.preventDefault();
    const { productDetails, category_id } = this.state;
    const { id } = this.props.details;
    if (Object.keys(this.props.details).length === 0) {
      this.props.addProduct(productDetails, category_id);
    } else {
      this.props.updateProduct(productDetails, category_id, id);
    }
    const timeoutId = setTimeout(() => this.props.getAllProducts(), 500);
    this.setState({ timeoutId });
  };

  handleFormChange = (event) => {
    const { productDetails } = this.state;
    const { name, value } = event.target;
    productDetails[name] = value;
    this.setState({
      productDetails
    });
  };

  handleSelectChange = (event, { value }) => {
    this.setState({
      category_id: value,
      category_name: event.target.textContent
    });
  };

  firstThreeFields = () => {
    const { successResponse } = this.props.allCategories;
    const { image_url } = this.state.productDetails;
    const { category_id, category_name } = this.state;
    const dropdownCategories = successResponse.map(({ name, id }) => ({ text: name, value: id }));
    return (
      <Fragment>
        <Form.Field className="formField">
          <label>Product Name</label>
          <input
            placeholder="Product Name"
            value={this.state.productDetails.name}
            onChange={this.handleFormChange}
            name="name"
          />
        </Form.Field>
        <Form.Field className="formField">
          <label>Category</label>
          <Select
            placeholder="Select a Category"
            onClick={this.getAllCategories}
            text={category_name}
            value={category_id}
            options={dropdownCategories}
            onChange={this.handleSelectChange}
            name="category_id"
          />
        </Form.Field>
        <Form.Field className="formField">
          <label>Image URL</label>
          <input
            placeholder="Image URL (example:https://exampleimageurl.jpg)"
            typeof="url"
            value={image_url}
            onChange={this.handleFormChange}
            name="image_url"
          />
          <a target="_blank" rel="noopener noreferrer" href="https://imgur.com/">
            Upload image here and provide link
          </a>
        </Form.Field>
      </Fragment>
    );
  };

  lasthreeFields = () => {
    const { description, quantity, price } = this.state.productDetails;
    return (
      <Fragment>
        <Form.Field className="formField">
          <label>Product Price</label>
          <input
            placeholder="Price"
            typeof="number"
            value={price}
            onChange={this.handleFormChange}
            name="price"
          />
        </Form.Field>
        <Form.Field className="formField">
          <label>Quantity</label>
          <input
            placeholder="Quantity"
            typeof="number"
            value={quantity}
            onChange={this.handleFormChange}
            name="quantity"
          />
        </Form.Field>
        <Form.TextArea
          placeholder="Short Description"
          className="formField"
          value={description}
          onChange={this.handleFormChange}
          name="description"
        />
      </Fragment>
    );
  };

  render() {
    return (
      <Form className="modalForm" onSubmit={this.handleFormSubmit}>
        {this.firstThreeFields()}
        {this.lasthreeFields()}
        <Button type="submit">Submit</Button>
      </Form>
    );
  }
}

const mapStateToProps = state => ({
  allCategories: state.allCategories
});

export { ProductForm as ProductFormPage };
export default connect(
  mapStateToProps,
  { productCategories, updateProduct, addProduct, getAllProducts }
)(ProductForm);
