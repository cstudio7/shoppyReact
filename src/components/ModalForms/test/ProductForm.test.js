import React from 'react';
import { shallow } from 'enzyme';
import { ProductFormPage } from '../ProductForm';
import { productCategories } from '../../../actions/getCategoriesAction';
import { updateProduct } from '../../../actions/updateProductAction';
import { getAllProducts } from '../../../actions/getProductsAction';
import { addProduct } from '../../../actions/addProductAction';

describe('Test the ProductForm Component', () => {
  it('should match snapshot', () => {
    const details = {};
    const allCategories = {
      successResponse: [{ id: 2, name: 'categoryname1' }, { id: 2, name: 'categoryname1' }]
    };
    const wrapper = shallow(
      <ProductFormPage
        allCategories={allCategories}
        details={details}
        productCategories={productCategories}
        updateProduct={updateProduct}
        getAllProducts={getAllProducts}
        addProduct={addProduct}
      />
    );
    const event = {
      target: {
        name: 'name',
        value: 'pname'
      },
      preventDefault: () => {}
    };
    const value = 5;
    wrapper.instance().getAllCategories();
    wrapper.instance().handleFormChange(event);
    wrapper.instance().handleFormSubmit(event);
    wrapper.instance().handleSelectChange(event, { value });
    wrapper.instance().componentWillUnmount();
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot', () => {
    const details = {
      nameToDisplay: 'John Doe',
      imageUrl: 'https://i.imgur.com/27DUH5b.jpg',
      description: 'admin@mail.com',
      productPrice: 300,
      productQuantity: 3
    };
    const allCategories = {
      successResponse: [{ id: 2, name: 'categoryname1' }, { id: 2, name: 'categoryname1' }]
    };
    const wrapper = shallow(
      <ProductFormPage
        allCategories={allCategories}
        details={details}
        productCategories={productCategories}
        updateProduct={updateProduct}
        getAllProducts={getAllProducts}
        addProduct={addProduct}
      />
    );
    const event = {
      target: {
        name: 'name',
        value: 'pname',
        textContent: 'textXontent'
      },
      preventDefault: () => {}
    };
    const value = 9;
    wrapper.instance().getAllCategories();
    wrapper.instance().handleFormChange(event);
    wrapper.instance().handleFormSubmit(event);
    wrapper.instance().handleSelectChange(event, { value });
    wrapper.instance().componentWillUnmount();
    expect(wrapper).toMatchSnapshot();
  });
});
