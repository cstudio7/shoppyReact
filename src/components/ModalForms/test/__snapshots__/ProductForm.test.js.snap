// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the ProductForm Component should match snapshot 1`] = `
<Form
  as="form"
  className="modalForm"
  onSubmit={[Function]}
>
  <FormField
    className="formField"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Product Name"
      value="pname"
    />
  </FormField>
  <FormField
    className="formField"
  >
    <Select
      name="category_id"
      onChange={[Function]}
      onClick={[Function]}
      options={
        Array [
          Object {
            "text": "categoryname1",
            "value": 2,
          },
          Object {
            "text": "categoryname1",
            "value": 2,
          },
        ]
      }
      placeholder="Select a Category"
      value={5}
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="image_url"
      onChange={[Function]}
      placeholder="Image URL (example:https://exampleimageurl.jpg)"
      typeof="url"
      value=""
    />
    <a
      href="https://imgur.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Upload image here
    </a>
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="price"
      onChange={[Function]}
      placeholder="Price"
      typeof="number"
      value=""
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="quantity"
      onChange={[Function]}
      placeholder="Quantity"
      typeof="number"
      value=""
    />
  </FormField>
  <FormTextArea
    as={[Function]}
    className="formField"
    control={[Function]}
    name="description"
    onChange={[Function]}
    placeholder="Short Description"
    value=""
  />
  <Button
    as="button"
    type="submit"
  >
    Submit
  </Button>
</Form>
`;

exports[`Test the ProductForm Component should match snapshot 2`] = `
<Form
  as="form"
  className="modalForm"
  onSubmit={[Function]}
>
  <FormField
    className="formField"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Product Name"
      value="pname"
    />
  </FormField>
  <FormField
    className="formField"
  >
    <Select
      name="category_id"
      onChange={[Function]}
      onClick={[Function]}
      options={
        Array [
          Object {
            "text": "categoryname1",
            "value": 2,
          },
          Object {
            "text": "categoryname1",
            "value": 2,
          },
        ]
      }
      placeholder="Select a Category"
      text="textXontent"
      value={9}
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="image_url"
      onChange={[Function]}
      placeholder="Image URL (example:https://exampleimageurl.jpg)"
      typeof="url"
      value="https://i.imgur.com/27DUH5b.jpg"
    />
    <a
      href="https://imgur.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Upload image here
    </a>
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="price"
      onChange={[Function]}
      placeholder="Price"
      typeof="number"
      value={300}
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="quantity"
      onChange={[Function]}
      placeholder="Quantity"
      typeof="number"
      value={3}
    />
  </FormField>
  <FormTextArea
    as={[Function]}
    className="formField"
    control={[Function]}
    name="description"
    onChange={[Function]}
    placeholder="Short Description"
    value="admin@mail.com"
  />
  <Button
    as="button"
    type="submit"
  >
    Submit
  </Button>
</Form>
`;
